crates/typst/src/math/accent.rs

```diff
@@ -1,18 +1,71 @@
 use crate::diag::{bail, SourceResult};
 use crate::foundations::{
-    cast, elem, Content, Packed, Resolve, Smart, StyleChain, Value,
+    cast, elem, func, Content, NativeElement, Packed, Resolve, Smart, StyleChain, Value,
 };
 use crate::layout::{Em, Frame, Length, Point, Rel, Size};
 use crate::math::{
     style_cramped, FrameFragment, GlyphFragment, LayoutMath, MathContext, MathFragment,
     Scaled,
 };
-use crate::symbols::Symbol;
 use crate::text::TextElem;
 
 /// How much the accent can be shorter than the base.
 const ACCENT_SHORT_FALL: Em = Em::new(0.5);
 
+/// This macro generates accent-related functions.
+///
+/// ```ignore
+/// accents! {
+///     '\u{0300}' | '`' => grave,
+/// //  ^^^^^^^^^    ^^^    ^^^^^
+/// //  |            |      |
+/// //  |            |      +-- The name of the function.
+/// //  |            +--------- The other characters that represent the accent.
+/// //  +---------------------- The default character that represents the accent.
+/// }
+/// ```
+///
+/// When combined with the `Accent::combine` function, accent characters can be normalized
+/// to the default character.
+macro_rules! accents {
+    ($($($ch:literal)|+ => $name:ident),+ $(,)?) => {
+        impl Accent {
+            /// Normalize an accent to a combining one.
+            pub fn combine(c: char) -> Option<char> {
+                Some(match c {
+                    $(
+                        $($ch)|+ => accents!(@__private first $($ch),+),
+                    )+
+                    _ => return None,
+                })
+            }
+        }
+
+        $(
+            /// The accent function for callable symbol definitions.
```

Comment:
Would be nice if this was customized for each symbol. Could be something like:
```rs
/// The accent function for
#[doc = stringify!($primary)]
/// .
```
(Something like this. Didn't test whether this works correctly...)